### DISCUSSION TITLE
bpftrace office hours?

### DISCUSSION THREAD

**Discussion by BurntBrunch** [bpftrace/bpftrace#2389] [Development]
I'm opening this to gauge interest in a semi-regular development-oriented office hours meeting? 

BPF office hours has been moderately successful and while we have a much smaller community here, I think it'd be nice to have a venue to meet and discuss topics in real time. For example, what people are working on short- and long-term, what roadblocks may exist for PRs, policy discussions, etc.

I personally would have found this useful initially and still have uses for it (and not only pinging for lands/reviews :stuck_out_tongue_closed_eyes:). 

@fbs, @viktormalik, @ajor, @danobi, you have the vast majority of contributions over the last two years, so your opinions count more!

**Comment by fbs** [comment]
Personally I do think it can be fun but I haven't done a lot for the project the past year(s?) so I doubt I add much value. It could be interesting to have a discussion on how get can move the project forward, decrease the latency on MRs and stuff

**Comment by dxuuu** [comment]
Hi Delyan. So I've been meaning to reply to this thread (funny how every time I think that I don't actually do it).

I think this is a good idea and would probably make development more fun. I've been less involved lately than I'd like so this would probably be a good motivator. There's been a bunch of things I've wanted to do (hello AOT scripts) but lost momentum on.

Not sure how timing would work out with the $DAYJOB but I can make a best effort attempt to show up.

**Comment by viktormalik** [comment]
Hi, this is a great idea, I'm certainly in. Especially recently, we've been falling behind with PR reviews, so this could speed things a little bit up. And could make the development more fun :-)

**Comment by BurntBrunch** [comment]
Great, let's figure out a time and I'll post a draft agenda to add to! (I'm thinking a 30 min slot to start with.)

Happy to work with people's time zones - can you give us a few time slots that work for you?
I'm in PST but I know a few people are GMT/east of GMT, so I'm happy with anything early morning - 6am+ PST is fine (2pm+ BST).

**Reply by viktormalik** [reply]
I'm in CET (GMT+1 in winter, GMT+2 in summer). I can do Tuesday any time until 5pm, (usually) Wednesday 4-5pm, or Friday 3-4pm.

**Reply by danobi** [reply]
I'm in MT (UTC-7 / UTC-6). After 6pm local time or before 10am local works best for me. But I am happy to be flexible

**Reply by fbs** [reply]
Same zone as @viktormalik, (amsterdam region)

**Comment by brendangregg** [comment]
I'm Sydney time (GMT+11ish) although I work odd hours to overlap with the US: Tues to Sat, 7am to 3pm. If it's not between midnight and 6am (1/4th of the clock) there's certainly a chance I can join. My main interest is tool development and capabilities.

**Comment by BurntBrunch** [comment]
Thanks, everyone, here are some time slots that match the constraints (@ajor, I think some of these would work for you but feel free to request changes):

1. Tuesdays (e.g. Nov 1) at: 8am PDT (me), 9am MDT (@danobi), 4pm CET (@fbs, @viktormalik), 2am AEDT 
  a. Sorry @brendangregg, the constraints put Australia straight in the midnight-6am bucket, there's no time that works reasonably for Europe, Australia, and North America :frowning_face: We can maybe do separate calls and capture notes?
2. Wednesdays (e.g. Nov 2) at the same times as above.
3. Fridays (e.g. Nov 4) at: 7am PDT (me), 8am MDT (@danobi), 3pm CET (@fbs, @viktormalik), 1am AEDT

We should probably have the first meeting be 45 mins for intros and so on.
Logistically, we can use Zoom on a free account or the Meta instance (I have to check but we may able to record it there), or BBB if anyone has an instance.

:arrow_forward: **Please vote for a day of the week and let me know if you can do next week or the week after.**

Agenda-wise, I have the following items:

* Work in flight summary and path forward for myself
* Path to cleaning up the queues
  * Set up a policy (it can be automated via Actions) for auto-closing of issues and PRs
  * Enable auto-merge
* Path to easier onboarding of new contributors
  * Tests seem to be a common source of confusion, maybe a revamped CONTRIBUTING.md?
* Better CI (if we have time)

:arrow_forward: **Again, please let me know what day of the week and first date you can do, as well as any agenda topics you may have!**

**Reply by fbs** [reply]
Next wednesday at 1600 CET or friday 1500 CET should work for me. After that we have to see, work meetings have to come first

**Reply by fbs** [reply]
As for how, https://meet.jit.si/ might be a good option

**Reply by brendangregg** [reply]
> Thanks, everyone, here are some time slots that match the constraints (@ajor, I think some of these would work for you but feel free to request changes):
> 
> 1. Tuesdays (e.g. Nov 1) at: 8am PDT (me), 9am MDT (@danobi), 4pm CET (@fbs, @viktormalik), 2am AEDT
>    a. Sorry @brendangregg, the constraints put Australia straight in the midnight-6am bucket, there's no time that works reasonably for Europe, Australia, and North America ☹️ We can maybe do separate calls and capture notes?

We do the BPF technical steering committee meeting across those three zones: Next one is 8am AEDT, 2pm PDT, 3pm MDT, 11pm CET. Could go back an hour. (Although mid year that slot is 6am AEDT, 1pm PDT, 2pm MDT, 10pm CET.) It puts India in the middle of the night, though, so we originally rotated between two time slots to get world coverage (not necessary for the current BSC makeup). I'd consider doing that here -- a rotating timeslot.

Also note some context: Some tech companies are pushing staff to come back to their Silicon Valley offices. I wouldn't want to create friction for remote work.

**Reply by viktormalik** [reply]
I'm ok with all the above 3 options, although might not be able to join on Wed, Nov 2.

**Reply by danobi** [reply]
All 3 times work for me. And I currently don't have any recurring work meetings for those times.

**Reply by ajor** [reply]
Any weekday would be good for me. Can make next week or later.

**Reply by BurntBrunch** [reply]
@brendangregg, the conflicting daytime constraints in both NA and European time zones make that time slot unavailable to 2-3 people.

We can certainly make an A/B schedule where we rotate to capture Sydney too but given that all recent contributors can make the above slots, I think we should go with one of them as the first meeting.

**Comment by danobi** [comment]
One topic in the back of my mind for a while now: merging the asciidoc man page and `reference_guide.md`.

**Comment by BurntBrunch** [comment]
Okay, unless there's serious objections, let's say the date and time for the first meeting is:
 
 Tuesday Nov 1 at: 8am PDT (me), 9am MDT (@danobi), 3pm BST (@ajor), 4pm CET (@fbs, @viktormalik)
 Duration: 45 min
 
 This is the initial agenda just so we have a starting point, we can play it by ear:

1. Intros & top-of-mind work (10-ish mins)
2. Better docs (asciidoc -> reference_guide.md)
3. Queue cleanup (automation, auto-merge, etc)
4. Easier contributing (better docs, dev environment, CI, etc)

Link: https://meet.jit.si/moderated/cfcf1ea615c4dec1adbb5bd68aca2f9d687e0373c4a65cafcc66b95d4ed9a3f5

Zipped ics invite if you want it: [bpftrace office hours.zip](https://github.com/iovisor/bpftrace/files/9873536/bpftrace.office.hours.zip) 
(I'm also happy to email the invite to people, not sure if people actively use their github emails)

**Comment by xh4n3** [comment]
Hi guys, is this gonna be open to anyone, or it is kind of exclusive to core members? I really wish I could know what's going on in the community 😃

**Reply by danobi** [reply]
I definitely think it should be open to everyone.

**Reply by xh4n3** [reply]
That's great!

**Reply by BurntBrunch** [reply]
Given that it's all arranged on this thread, yeah, it's very much open to anyone :)

**Comment by BurntBrunch** [comment]
Summary of today's meeting (thanks for joining, everyone!):

Re: documentation and asciidoc:
  * People generally update the reference guice or the man page but not both
  * The man page tries to be more in-depth than the reference guide
  * Merging them would save time and make them more consistent, ideally generated from the same docs
  * We should ideally host it at bpftrace.org (@ajor also wants a new logo or at least not that logo!)
  * @danobi will take a look at unifying the docs

Re: issues auto-closing
  * Consensus is that it's a bad idea

Re: PRs auto-closing for staleness
  * Maybe it's a good idea? Main argument is what it says about contribution velocity and state of maintenance.

Re: code owners file and getting tagged on PRs
  * It was generally viewed as a good idea. The idea of the tooling picking one person to tag instead of all was floated but there's very few reviewers at the moment and it's not clear that would work better for bpftrace.

Re: auto-merging on green CI
  * There were questions raised about whether you can approve but also request a mandatory review from someone else before the PR is merged.
  * **Follow-up**: The state of [github's auto-merge/PR review integration](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request) is as follows:
    * You can specify branch protection rules (CI is green and N approving reviews from people with `write` or `admin` bits) that are on by default. Auto-merge will trigger when all rules pass.
    * "Request changes" from reviewers can be bypassed by people with `admin` or `write` bits in order to speed things up if the reviewer who requested changes is not responsive.
    * You can disable auto-merge on an individual PR basis. 
      * @ajor I think your concern about asking for more reviews is covered by this. Disable auto-merge for that PR, wait for more reviews, and the last reviewer can re-enable it or merge it directly.

Re: contribution bar is very high
  * Maybe more pain should be tolerated on master in exchange to faster PR velocity?
  * CI is a nightmare - flakiness, debugging, all of it.
    * Can we set it up so people can restart their CI on their own PRs?
    * **Follow-up**: The answer is no, [you need the `write` permission](https://docs.github.com/en/actions/managing-workflow-runs/re-running-workflows-and-jobs) to re-run jobs.

Re: next meeting
  * Cadence at once a month seems acceptable.
  * Should capture the other side of the globe.
  * Some time before the holidays.
  * I propose one of these time slots, depending on whether the GMT/CET folks want to join very early in their morning:
    * Tue Dec 6 8pm PST, 9pm MST, (Wed Dec 7) 3pm AEDT (middle of the night for GMT/CET)
    * Tue Dec 6 10pm PST, 11pm MST, (Wed Dec 7) 6am GMT, 7am CET, 5pm AEDT

**Reply by xh4n3** [reply]
Regarding the CI part, there are a few Github Actions can achieve what we need, maybe consider enabling them.
https://github.com/marketplace/actions/rerun-actions
https://github.com/marketplace/actions/close-stale-issues

**Reply by viktormalik** [reply]
Re: next meeting:
- 7am CET is completely fine with me, the other one (5am if I count correctly) would be more challenging (not impossible, though).

Re: contribution bar is very high:
- We mentioned several times that we should document "tips and tricks" for contributing. We already have some [contribution guidelines](https://github.com/iovisor/bpftrace/blob/master/docs/developers.md), I'll look into them, expand them with what we discussed, and make them more visible.

**Reply by BurntBrunch** [reply]
> Regarding the CI part, there are a few Github Actions can achieve what we need, maybe consider enabling them. 

`rerun-actions` is maybe a good idea but doesn't help brand new contributors, unless we start tagging every PR with `ok-to-test`. Hm, need to think about this more.

> I'll look into them, expand them with what we discussed, and make them more visible.

Thank you! I can document my dumb `distrobox` workflow too (the one where you delete the entrypoint from the Dockerfile, so the image is usable by distrobox).

**Comment by BurntBrunch** [comment]
Hi folks, I got laid off today, so my involvement here is probably not going to continue. I have a lot of ideas and maybe I'll find a company who'd hire me to work on them. Until then, however, I think I'll step away from tech for a bit.

Sorry for the wild roller coaster ride.

**Reply by viktormalik** [reply]
So sorry to hear that Delyan! Wish you all the best in your next journey and if you ever start missing tech, we'll be more than happy to have you here :-)

And thanks for bringing the fresh energy to this project!

**Comment by viktormalik** [comment]
Hi @fbs, @ajor, @danobi (and others, too)!

Unfortunately, since Delyan left, the maintainer activity in this project has dropped significantly. People are still opening issues and PRs, though, and while I'm able to review most of them at the moment (with @fbs responding to issues and questions, thanks for that!), there's a couple of tasks that I'd need help from other maintainers. The most important would be to review my PRs, there's already several of them and I don't want to merge anything without a proper review.

I'm aware that all of you have bpftrace as a side-project that you do in your free time so I just wanted to ask if I there's anything I could do to help getting you back (if you want to come back to this project at all). I felt like these regular meetings were a good mean to stay involved in the community, so I would gladly take over their organization, if you think that they would add some value.

Thanks for the opinions!

**Reply by fbs** [reply]
yeah lets have another meeting :)

**Reply by dxuuu** [reply]
Hi sorry, I missed the notification. I'm open to more meetings. I'll try to start making regular passes on the PRs again

**Reply by ajor** [reply]
More meetings sound good to me :)

I'm also going to be doing some work on bpftrace as part of my job now, so I'll be more active here on GitHub in the future too

**Comment by viktormalik** [comment]
Great, so let's have a meeting the next week. 

### Time slot

If you timezones haven't changed, I suggest one of these:
- Option 1: Monday (April 3) 9am MDT (@danobi), 4pm BST (@ajor), 5pm CEST (me and @fbs)
- Option 2: Tuesday (April 4) 8am MDT (@danobi), 3pm BST (@ajor), 4pm CEST (me and @fbs)
- Option 3: Wednesday (April 5) 9am MDT (@danobi), 4pm BST (@ajor), 5pm CEST (me and @fbs)

Let me know if any of these suits you. I personally prefer options 2 and 3.

### Agenda

So far, I have 3 points in mind:

1. Queue cleanup - what can we do better?
2. Overview of significant ongoing work
a. #2334: we should continue with this initiative
b. Getting rid of `->` (#2477)
3. Contribution bar (follow-up from the last meeting).

Feel free to suggest any more.

**Reply by ajor** [reply]
All times are good for me!

**Reply by fbs** [reply]
I  cant do the wednesday

**Reply by danobi** [reply]
All three work for me

**Reply by viktormalik** [reply]
Great, so let's meet on Tuesday, April 4th, 4pm CEST (8am EDT, 3pm BST).

Link: https://meet.google.com/jsk-rrse-srn  
Invitation: [23-04-04_bpftrace_office_hours.tar.gz](https://github.com/iovisor/bpftrace/files/11134985/23-04-04_bpftrace_office_hours.tar.gz)

The meeting is open to anyone :-) (ping @xh4n3 as he joined the last time)

**Reply by fbs** [reply]
I'm afraid I have to cancel :(. Got a last minute family thing pop up. Don't have much to add, just happy that @ajor has time to do work.

**Comment by viktormalik** [comment]
I created a document for the office hours meeting minutes:
https://docs.google.com/document/d/1nt010RfL4s4gydhCPSJ-Z5mnFMFuD4NrcpVmUcyvu2E/edit?usp=sharing

I also opened #2550 which adds the link to the README.

I'd suggest keeping this discussion open to settle on meeting times if necessary.

**Comment by viktormalik** [comment]
The time for the next meeting is approaching, any updates in timezones?
@ajor, @fbs, @danobi.

**Reply by ajor** [reply]
I'm good for Tuesday 2nd - Friday 5th next week, GMT.

**Reply by danobi** [reply]
Tues - Fri also works for me. Still MT

**Reply by fbs** [reply]
I can't do tmorrow (tue), but should be ok for the rest of the week :)

**Reply by viktormalik** [reply]
Ok, let's do Wednesday, May 3rd, 9am MT, 4pm BST, 5pm CEST. I'll put the information, including the meet link, to the document.

**Reply by viktormalik** [reply]
I added meeting minutes from yesterday to the doc.

One more organizational thing: @fbs and @ajor, if you have a Google account, could you send it to me? I'd add you to the document and the meeting so that you can edit and join them without my interception.

Thanks!

**Reply by ajor** [reply]
Could you add my work email please? It's my Git Hub username at meta.com

**Comment by viktormalik** [comment]
Hi, sorry for a bit longer interval from the last meeting, I had a lot on my plate recently. Anyways, I'm able to do a meeting starting from the next week. What is your availability @fbs, @ajor, @danobi?

I'd also love to get @brendangregg to the meeting one day so that we could discuss things around tools. If you had time to join, we'd be able to move the time so that it suits you.

**Reply by ajor** [reply]
My calendar's pretty full on Tuesday, but the rest of the days next week are looking ok for me

**Reply by brendangregg** [reply]
Yes, I'd like to meet; although I'm in Singapore for SREcon, and am travelling next week. I'm back to normal June 26th onwards. I can usually do meetings between 7am and midnight Sydney time.

**Reply by danobi** [reply]
Pacific coast hours any time next week would work for me. June 26th+  would probably work better for me too. Currently still out of country (but still working).

**Reply by alban** [reply]
Hello! I'd be interested in joining the next meeting. I'm in the Berlin timezone.

My topics of interests are related to [Inspektor Gadget](https://github.com/inspektor-gadget/inspektor-gadget)'s needs:
- bpftrace using externally defined bpf functions (freplace) for filtering events.
- adding custom fields to all printf calls (e.g. timestamp, namespace/cgroup ids), regardless of the user script passed with `-e`.

cc @mauriciovasquezbernal

**Reply by ajor** [reply]
I'm available every day on the week of the 26th

**Reply by mmarchini** [reply]
I may be interested in joining depending on time and day

**Reply by viktormalik** [reply]
Let's do it in the week of the 26th. There aren't many good time options as someone will always need to join early morning or late evening. These two look at least a bit doable:

1. 6am BST (@ajor), 7am CEST (@fbs, @viktormalik, @alban), 3pm AEST (@brendangregg), 10pm PDT (@danobi?)
2. 2pm BST (@ajor), 3pm CEST (@fbs, @viktormalik, @alban), 11pm AEST (@brendangregg), 6am PDT (@danobi?)

I can do option one any day except Wednesday 28th and option 2 any day except Friday 30th.

**Reply by danobi** [reply]
I can do either option the entire week except:
* (2) on 6/28
* (1) on 6/30

---

Since it looks like the OH are getting more attendance (which is great!), I would suggest either using some kind of scheduling app (eg https://calendly.com/ which IIRC works fairly well) or picking ~2 times and switching between them. Preset seems like less of a administration hassle -- I will just make sure I don't plan anything during those times in the future.

**Reply by viktormalik** [reply]
Good point, let's come up with 2 times that work for most timezones and switch between them. I'll put it to the agenda of the next meeting.

**Reply by viktormalik** [reply]
Since there seem to be no more conflicts, let's do option (1) on Thursday 29th (Wednesday 28th for US). I created a GCal event, I added everyone whose email I had. If someone else wants an invitation, send me your email address.

The GMeet link stays the same and can be also found in the meeting minutes: 
https://meet.google.com/joo-jzcy-cqk

**Reply by mmarchini** [reply]
@viktormalik can you also add me? oss@mmarchini.me

**Reply by brendangregg** [reply]
Either of the times work for me thanks; please add me to the event so it shows up on my calendar.

**Reply by viktormalik** [reply]
@brendangregg which email should I use?

**Comment by viktormalik** [comment]
Hopefully this is the last thread in this discussion :-)

I checked possible regular times for bpftrace office hours and the one we selected is great as it covers pretty much every reasonable timezone. I'd still prefer to alter between two times to allow joining for people who have a regular conflict in one of the times and also to be more welcoming to people from the US west coast. We could either shift the other time by 2-3 hours forward or to select something completely different.

To make the selection simpler, I created a Doodle poll for the first week of August. Please do select all times that are suitable for you and I'm going to pick the two most voted and we'll switch between them periodically (every time the first week of the month).

https://doodle.com/meeting/participate/id/dwrEYxwb

**Comment by viktormalik** [comment]
August is approaching so I made a decision on the regular schedule. I also put it down to the document.

We will switch between two times:
- odd months: the first Tuesday, 5pm CET/CEST
- even months: the first Thursday, 3pm CET/CEST

This means that the next meeting will be on **Thursday, August 3rd, 3pm CEST**.

The meeting stays the same: https://meet.google.com/joo-jzcy-cqk.

I will also create recurring events in the calendar and invite all those that attended the last meeting.

@ajor, @danobi, @fbs, @brendangregg, @mmarchini, @alban, @xh4n3

**Comment by viktormalik** [comment]
This is a kind reminder that the next office hours meeting takes place today at 5pm CEST.

**Reply by fbs** [reply]
woop sorry, missed this in my calender, can't make it :(

**Reply by danobi** [reply]
Ah crap I gotta figure out how to stop forgetting

**Reply by viktormalik** [reply]
No worries, I updated the meeting minutes with today's meeting.

The next meeting will be on **October 5, 3pm CEST**. I'll send a reminder through this discussion the day before.

**Comment by brendangregg** [comment]
thanks, finally got it to show up on my calendar; I won't be able to make this one though.

**Comment by viktormalik** [comment]
This is a kind reminder that the next office hours meeting will happen this week on **October 5, 3pm CEST**.

Please fill items to the agenda.

**Comment by viktormalik** [comment]
Reminder: the office hours meeting is starting now

**Comment by viktormalik** [comment]
This is a kind reminder that the next office hours meeting will happen tomorrow on **November 7, 5pm CET**.

Please fill items to the agenda.

**Comment by viktormalik** [comment]
Reminder: the office hours meeting starts in 10 minutes.

**Comment by viktormalik** [comment]
This is a kind reminder that the next office hours meeting will happen tomorrow on December 7, 3pm CET (2pm UTC).

Please fill items to the agenda.

**Comment by viktormalik** [comment]
Reminder: the office hours meeting starts in 20 minutes.

**Comment by viktormalik** [comment]
Hi, the next office hours meeting is scheduled for Jan 2, which is the day after New Year's Eve. I personally planned to take some time off for that day, what about if we rescheduled this one to Jan 4 (Thursday same week), 5pm CET?

@ajor @danobi @fbs @jordalgo

**Reply by jordalgo** [reply]
Works for me.

**Reply by danobi** [reply]
Should be good for me as well

**Reply by ajor** [reply]
Thanks! I'm also having some time off then

**Reply by viktormalik** [reply]
Thanks! Rescheduled to Jan 4, 5pm CET.

**Comment by viktormalik** [comment]
Hi, the next office hours is scheduled for today, 5pm CET. We only have one item on the agenda so far, so please add more.

**Comment by viktormalik** [comment]
This is a kind reminder that the next office hours meeting will happen tomorrow, February 1, 3pm CET (2pm UTC).

Please fill items to the agenda.

**Reply by jordalgo** [reply]
Agenda doc: https://docs.google.com/document/d/1nt010RfL4s4gydhCPSJ-Z5mnFMFuD4NrcpVmUcyvu2E/edit#heading=h.v1c6uuvyxuw6

**Comment by viktormalik** [comment]
Hi, the next office hours meeting will take place today, April 4, 3pm CEST (1pm UTC).

Please, fill items to the agenda:
https://docs.google.com/document/d/1nt010RfL4s4gydhCPSJ-Z5mnFMFuD4NrcpVmUcyvu2E/edit?usp=sharing

**Comment by viktormalik** [comment]
Hi, the next office hours meeting will take place tomorrow, May 7, 5pm CEST (3pm UTC).

Please, fill items to the agenda:
https://docs.google.com/document/d/1nt010RfL4s4gydhCPSJ-Z5mnFMFuD4NrcpVmUcyvu2E/edit?usp=sharing
